tools/shared/src/main/scala/org/scalajs/core/tools/optimizer/IRChecker.scala

```diff
@@ -1060,9 +1060,16 @@ class IRChecker(unit: LinkingUnit, logger: Logger) {
 }
 
 object IRChecker {
-  private final class ErrorContext private (pos: Position, name: String) {
-    override def toString(): String =
+  private final class ErrorContext private (val treeOrLinkedClass: Any)
```

Comment:
I see. So I guess leaving this is fine. But please add a comment stating that this is optimized for allocations in the non-failure case (i.e. if `toString` is not called).